### ISSUE TITLE
Topic has unread comments if navigated to previous pages

### ISSUE THREAD

**Issue by cryptogun** [nitely/Spirit#163]
When I've read all the comments of a topic and navigated back to page 1, I found that this topic is red highlighted on topics list.
Is it better to implement update_bigger_or_create then [udpate_or_create](https://github.com/nitely/Spirit/blob/master/spirit/topic/utils.py#L15)?
And, there're many POST request to `/forum/comment/bookmark/1/create/`. Maximum 20 POSTs per topic page.

**Comment by nitely** [comment]
This has been brought up before. The bookmark work that way, if you go back then it bookmarks a previous comment. Like a real life bookmark. There is an easy fix by filtering `comment_number__lt=comment_number` [in the bookmark form](https://github.com/nitely/Spirit/blob/master/spirit/comment/bookmark/forms.py#L25).

> Is it better to implement update_bigger_or_create then udpate_or_create?

That gets called if you click a previous page, which is the intended behaviour. If you go back (ie: click on the web browser back button) then it does not get called (the web browser caches the page), but the form above does.

It should probably be updated if it's lesser than the stored comment number as you suggest.

> And, there're many POST request to /forum/comment/bookmark/1/create/. Maximum 20 POSTs per topic page.

There is no workaround that.

**Comment by cryptogun** [comment]
> It should probably be updated if it's lesser than the stored
No, I don't want it to be lesser actually. I don't want _a real life bookmark_ cos I think that would confuse our forum user.
So my modifications:
- `comment_number__lt=comment_number`. Hence 2, 3, 4, 5 ... no backward.
- [These line](https://github.com/nitely/Spirit/blob/master/spirit/comment/bookmark/models.py#L59-L63) to
```python3
        try:
            obj = cls.objects.get(user=user, topic=topic)
        except cls.DoesNotExist:
            obj = cls.objects.create(user=user, topic=topic, comment_number=0)

        if comment_number > obj.comment_number:
            obj.comment_number = comment_number

        obj.save()
```
Hence 1, 21, 31, 41 ... no backward.
> many POST request
Not big deal to me for now. If anyone want to handle this and don't need `real life bookmark`(readed is readed, no need to bookmark previous page) like me, they can remove the corresponding js script and change `topic_viewed()` to bookmark first comment of the next page.
